### ISSUE TITLE
Bug: [no-restricted-types] Does not flag `Omit` when used in an `interface` `extends`

### ISSUE THREAD

**Issue by simon-abbott** [typescript-eslint/typescript-eslint#12114]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.3&fileType=.tsx&code=JYOwLgpgTgZghgYwgAgOIHt0BNkG8BQyyMmAXMgM5hSgDmA3IcgEZxTlU0gP4C%2B%2B%2BUJFiIUAITg4IAD0ggsFZAHkAtsDAAeDNgA0yAOQl0%2BgHx4mR8iACuK5tEb98YAJ4AHcZIBMyALzK1TW0sPUNMU3ogA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uhXyrPyIAJrSKlUGANrhsORNGgdokADQzZWWdkhiU6TOq04A8gFtK%2BfZABCAQyEBCSIdkBfF2HeyAujPeugA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
interface Good {
  foo: string;
  bar: string;
}

interface Bad extends Omit<Good, 'foo'> {
  foo: number;
}

type Bad2 = Omit<Good, 'foo'>;
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-restricted-types": [
      "error",
      {
        "types": {
          "Omit": "Bad!"
        }
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I would expect that both instances of `Omit` would be flagged by the rule.

### Actual Result

The `Omit` inside the `interface ... extends` clause is not flagged.
